### DISCUSSION TITLE
Login not working

### DISCUSSION THREAD

**Discussion by rodriguesabner** [MasterKale/SimpleWebAuthn#206] [Troubleshooting]
hey, I did the implementation here but when I login the data does not appear. Do you know what it could be?

repo: https://github.com/rodriguesabner/webauthn-server

![D23B8C96-7395-4C84-AE0C-83B808EA0878](https://user-images.githubusercontent.com/40338524/169889457-ab62a93b-6ae8-4380-8846-9e3e300cc4ac.png)

**Comment by MasterKale** [comment]
Hello @rodriguesabner, thanks for posting. I took the liberty of converting this into a Discussion item.

First of all, since I can't read the Portguese, can I ask you to please translate the message in the screenshot above? I can guess from prior experience it's prompting you to use a security key to authenticate? Or is this still registration? I'll assume authentication for now.

One thing that'll help is an example of the value, in the browser, that you're feeding to `navigator.credentials.get()` (whether it's directly, through @simplewebauthn/browser's `startRegistration()`, etc...) Maybe it's a case of missing transports from entries in `allowCredentials`...

**Reply by rodriguesabner** [reply]
Hi @MasterKale, thanks for the answer.

That's it! I was sending the 'transports' null instead of an empty (or filled) array, now it finds me on authentication!

Now just one more question, my challenge is coming different from what is registered, do you know how I can get around this? 

In generateAuthenticationOptions I send the challenge, but it doesn't set:

```
         const options = generateAuthenticationOptions({
            timeout,
            challenge: existentUser.challenge,
            allowCredentials,
            userVerification,
            rpID
        });
```

**Reply by rodriguesabner** [reply]
Now it's working, with the same code lol! Thank you for your help!
